webapp/src/serial.tsx

```diff
@@ -84,6 +83,15 @@ export class Editor extends srceditor.Editor {
         }
     }
 
+    simStateChanged() {
+        // We're leveraging the chart's `nonRealtmeData` flag to get both smooth scrolling
+        // and the ability to pause the animation.
+        // When the chart is in realtime mode, scrolling is very smooth but cannot be paused.
+        // In non-realtime mode, scrolling only occurs when new data arrives, but scrolling
+        // is choppy (being tied to new data arriving).
+        this.charts.forEach((chart) => chart.setRealtimeData(this.parent.isSimulatorRunning()));
```

Comment:
We should only pause charts if they are coming from the simulator.

You can also have charts that come from the hardware (using serial over WebUSB)

Reply:
Oh nice catch. How can I test this? Will web search it.

Reply:
Create a Microbit script that uses plotBarGraph, flash it to the device and you’ve a second console button for device data.

Reply:
It isn't clear to me how to connect the running microbit program to the serial console. Do I need to pair the device? It isn't being detected when I try to pair it, but I can flash it by copying the hex file.

Reply:
Ok after updating the microbit firmware I was able to pair it and get the serial stream.

Reply:
Instructions here if you need it: https://makecode.microbit.org/device/usb/webusb